keanu-project/src/main/java/io/improbable/keanu/tensor/dbl/Nd4jDoubleTensor.java

```diff
@@ -472,6 +479,23 @@ public DoubleTensor logInPlace() {
         return this;
     }
 
+    /**
+     * This is identical to log().times(y), except that it changes NaN results to 0.
+     * This is important when calculating 0log0, which should return 0
+     * See https://arcsecond.wordpress.com/2009/03/19/0log0-0-for-real/ for some mathematical justification
+     * @param y The tensor value to multiply by
+     * @return the log of this tensor multiplied by y
+     */
+    @Override
+    public DoubleTensor logTimesInPlace(DoubleTensor y) {
+        TensorValidator<Double, Tensor<Double>> validator = TensorValidator.thatChecksForNaN();
+        validator.check(this);
+        validator.check(y);
+        this.logInPlace().timesInPlace(y);
```

Comment:
I think you want to do `this = this.logInPlace().timesInPlace(y)` as the `timesInPlace` support broadcasting (times with different ranked tensors) which don't fulfil the `inPlace` contract

Reply:
done